server/src/api/accounts/accounts.router.ts

```diff
@@ -21,7 +21,11 @@ accountsRouter.get('/', async (req: Request, res: Response) => {
 accountsRouter.get('/search', async (req: Request, res: Response) => {
   try {
     const resp = await findByUsername(req.query.username as string);
-    const code = resp.status.success ? 200 : 404;
+    let c1 = resp.status.success ? 200 : 404;
+    if(resp.status.success === true && resp.data.length === 0){
+      c1 = 404;
+    }
+    const code = c1;
     res.status(code).send(resp);
```

Comment:
You could drop `c1` and just use `code`. The reassigning seems unnecessary

Reply:
The originals returned const and wasn't sure if that needed to be maintained, but [I found articles](https://medium.com/dailyjs/use-const-and-make-your-javascript-code-better-aac4f3786ca1) saying it was at the very least recommended.

Reply:
Generally yes but in this instance the scope that the variable is in is small and having multiple differently named variables for the same value makes it harder to read so I think leaving it has `let` would be fine. You could also extract a function like the article suggests which would also be good future proofing if add any more return scenarios in the future

Reply:
I'm really confused, did I not do that? One of the conditionals is a ternary operator/assignment, but it's otherwise the same, no?

Reply:
I'm not sure what you mean, the ternary operator is fine. I'm talking about doing either 
```
let code = resp.status.success ? 200 : 400;
if ( resp.status.success === true && resp.data.length === 0) {
    code = 404;
}
res.status(code).send(resp);
```
or like the article suggests for extracting a function like
```
const calculateCode = () => {
    if (resp.status.success === true) {
        if (resp.data.length === 0) {
            return 404;
        }
        return 200;
    }
    return 404;
}
const code = calculateCode();
```

Reply:
In both of these `c1` is dropped which simplifies things because there is one less variable to keep track of and there is no real security risks having `code` be mutable like in the first code block

Reply:
Alright I'll just get rid of const then